core/src/com/unciv/models/ruleset/Unique.kt

```diff
@@ -7,23 +7,45 @@ import com.unciv.logic.map.MapUnit
 import com.unciv.logic.map.TileInfo
 import com.unciv.models.stats.Stat
 import com.unciv.models.stats.Stats
-import com.unciv.models.translations.fillPlaceholders
-import com.unciv.models.translations.getPlaceholderParameters
-import com.unciv.models.translations.getPlaceholderText
-import com.unciv.models.translations.hasPlaceholderParameters
+import com.unciv.models.translations.*
 import com.unciv.ui.worldscreen.unit.UnitActions
 import kotlin.random.Random
 
-class Unique(val text:String){
-    val placeholderText = text.getPlaceholderText()
-    val params = text.getPlaceholderParameters()
+class Unique(val text: String){
     /** This is so the heavy regex-based parsing is only activated once per unique, instead of every time it's called
      *  - for instance, in the city screen, we call every tile unique for every tile, which can lead to ANRs */
+    val placeholderText = text.getPlaceholderText()
+    val params = text.getPlaceholderParameters()
     val stats: Stats by lazy {
         val firstStatParam = params.firstOrNull { Stats.isStats(it) }
         if (firstStatParam == null) Stats() // So badly-defined stats don't crash the entire game
         else Stats.parse(firstStatParam)
     }
+    val conditionals: List<Unique> = text.getConditionals()
+    
+    
+    // This function will get LARGE, as it will basically check for all conditionals if they apply
+    // This will require a lot of parameters to be passed (attacking unit, tile, defending unit, civInfo, cityInfo, ...)
```

Comment:
I think it's good. The tests need access to a many objects and should stay agnostic on how to determine _which_ specific instances these are. Any 'elegant' approach using collections of polymorphic interfaces or worse would only complicate matters without significant gain. Also, can't get more than twice the number of Civilopedia pages. If they get passed around very often, wrap in a data class (data for the flexibility of the copy method, and for the free deconstruction support).

```kotlin
data class DataForConditionalTest(val civ: CivilizationInfo? = null, val city: CityInfo? = null, val attacker: ICombatant? = null,...)
fun conditionalsApply(data: DataForConditionalTest) = conditionals.all { conditionalApplies(it, data) }
fun conditionalApplies(condition: Unique, data: DataForConditionalTest): Boolean {
    (civ, city, attacker) = data
    return when (condition.placeholderText) {
...
}

```